core/services/relay/evm/chain_reader.go

```diff
@@ -219,6 +220,41 @@ func (cr *chainReader) GetLatestValue(ctx context.Context, readName string, conf
 	return nil
 }
 
+func (cr *chainReader) GetLatestValueWithHeadData(ctx context.Context, readName string, confidenceLevel primitives.ConfidenceLevel, params any, returnVal any) (*commontypes.Head, error) {
```

Comment:
This won't give you a guarantee that the attached block is the same block that the data was read from. You'll have to return the same block height calculated in [EventBinding](https://github.com/smartcontractkit/chainlink/blob/cf50dd0ce46792a2c87aa7c79a7d60db49e951d1/core/services/relay/evm/read/event.go#L262) or [MethodBinding](https://github.com/smartcontractkit/chainlink/blob/cf50dd0ce46792a2c87aa7c79a7d60db49e951d1/core/services/relay/evm/read/method.go#L129)